KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/Server.java

```diff
@@ -1506,6 +1529,17 @@ private void commit() throws SolrServerException {
             }
         }
 
+        private void deleteDataSource(Long dsObjId) {
+            String dataSourceId = Long.toString(dsObjId);
+            String deleteQuery = "image_id:" + dataSourceId;
+            try {
+                // Get the first result. 
+                UpdateResponse updateResponse = solrCore.deleteByQuery(deleteQuery);
+            } catch (SolrServerException | IOException ex) {
+                logger.log(Level.SEVERE, "Error deleting content from Solr. Solr image id " + dataSourceId, ex); //NON-NLS
```

Comment:
As above for the DeleteDataSourceAction, low level classes should propagate exceptions for handling by the top level classes. E.g., here, how does the Server know what the performAction needs to do in its context of a menu selection in the UI by a user? These exceptions should propagate to the public deleteDataSource method above, which should propagate all errors, and so on, up the chain. The exceptions can be wrapped in other exceptions as needed.

Reply:
Better handle exceptions